### PR TITLE
Event Pictures with facebook Links

### PR DESCRIPTION
**Pull request by townsend3721**

### CODE REVIEW

File: app/uploaders/image_uploader.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  storage :file
+  # storage :fog
+
+  # Override the directory where uploaded files will be stored.
+  # This is a sensible default for uploaders that are meant to be mounted:
+  def store_dir
+    "uploads/#{model.class.to_s.underscore}/#{mounted_as}/#{model.id}"
+  end
+
+  # Provide a default URL as a default if there hasn't been a file uploaded:
```
Comment (aurashjalalian) [inline app/uploaders/image_uploader.rb]: please remove unneeded code.

Comment (aurashjalalian) [inline app/uploaders/image_uploader.rb]: Please keep the code clean and remove commented code.

File: app/views/events/index.html.erb [hunk 2]
```diff
@@ -5,6 +5,7 @@
     Click event to add your name to the guest list.
     <thead >
       <tr>
+        <th scope="col">Photo</th>
```
Comment (aurashjalalian) [inline app/views/events/index.html.erb:8]: Should this appear below name?

File: app/views/events/index.html.erb [hunk 3]
```diff
@@ -15,10 +16,10 @@
         <th scope="col">Actions</th>
       </tr>
     </thead>
-
     <tbody>
       <% @events.each do |event| %>
         <tr>
+          <td><%= image_tag event.image.thumb if event.image? %></td>
```
Comment (aurashjalalian) [inline app/views/events/index.html.erb]: The default to this shouldn't be to have nothing (it looks weird on the view). There should be something that shows up if they don't upload a photo. This is especially true in the view for show.

File: app/views/events/past_events.html.erb [hunk 4]
```diff
@@ -4,6 +4,7 @@
   <table class="table table-borderless table-hover">
```
Comment (aurashjalalian) [inline app/views/events/past_events.html.erb:4]: View the comments that I put for events, they may be relevant here.

File: app/views/events/show.html.erb [hunk 5]
```diff
@@ -41,6 +45,13 @@
   <% end %>
 </p>
 
+<p>
+  <strong>Share on Facebook</strong>
+    <%=link_to 'http://www.facebook.com/sharer.php?u='+request.fullpath, :target => :blank do %>
```
Comment (aurashjalalian) [inline app/views/events/show.html.erb:49]: This didn't work when I tried it?

image

https://www.facebook.com/sharer.php?u=/events/2

File: app/views/events/index.html.erb [hunk 6]
```diff
@@ -28,6 +29,9 @@
           <td><% event.guests.each do |guest| %>
                 <%= guest.firstname %> <%= guest.lastname%>,
               <% end %>
+          <%= link_to 'http://www.facebook.com/sharer.php?u='+request.fullpath, :target => :blank do %>
```
Comment (aurashjalalian) [inline app/views/events/index.html.erb:34]: This didn't work when I tried it?

image

https://www.facebook.com/sharer.php?u=/events/

**Review by aurashjalalian — CHANGES_REQUESTED** [review]

File: app/views/events/show.html.erb (hunk 5 above)
Comment (townsend3721) [reply app/views/events/show.html.erb:49]: This link won't work until the website has an actual URL

File: app/views/events/index.html.erb (hunk 6 above)
Comment (townsend3721) [reply app/views/events/index.html.erb:34]: See above comment

**Comment by aurashjalalian** [comment]
ALSO, please add the uploads (and any other file that gets touched) to .gitignore and remove all of the photos you uploaded to test.

File: app/views/events/index.html.erb (hunk 2 above)
Comment (townsend3721) [reply app/views/events/index.html.erb:8]: Tried it, but it looks a little weird.  I think it looks best without the "Photo" header.  Will push and show you what I mean in a few minutes

File: app/views/events/index.html.erb (hunk 3 above)
Comment (townsend3721) [reply app/views/events/index.html.erb]: Should events not allowed to be saved without a photo? Or should I just have a default image?

**Review by aurashjalalian — APPROVED** [review]

**Review by dantoth24 — APPROVED** [review]
Looks great, the placeholder image is a nice touch

**Pull request merged**